week_3/src/main/scala/org/epfl/12_extension_methods/ID.scala

```diff
@@ -0,0 +1,16 @@
+object UserID:
+
+  opaque type UserID = Long
+
+  def parse(string: String): Option[UserID] = string.toLongOption
+
+  extension (userID: UserID)
+    def value: Long = userID
```

Comment:
It is true that `toString` is available on opaque types, so `value` is not strictly necessary for printing. It will be necessary though as soon as I want to perform a database query (where I really need to extract the underlying `Long` value).

I thought printing a message would be a simpler example, but I agree that it does not justify why we want an extractor… I will think about it.